### PR TITLE
Added error for mass function when mass is out of range

### PR DESCRIPTION
**Pull request by elisachisari**
Addresses issue #536. Ready to review.

### CODE REVIEW

File: src/ccl_massfunc.c [hunk 1]
```diff
@@ -573,6 +573,12 @@ double ccl_massfunc(ccl_cosmology *cosmo, double halomass, double a, double odel
 	  ccl_cosmology_set_status_message(cosmo, "ccl_background.c: ccl_cosmology_compute_growth(): Support for the halo mass function in cosmologies with massive neutrinos is not yet implemented.\n");
 	  return NAN;
   }
+
+  if (halomass > 1e17 || halomass < 1e6){
```
Comment (damonge) [inline src/ccl_massfunc.c]: I think these should be set to `spline_params.LOGM_SPLINE_MIN` and `spline_params.LOGM_SPLINE_MAX`
@villarrealas do you agree?

Comment (villarrealas) [reply src/ccl_massfunc.c]: I forgot that we were making those user accessible. I agree that this change should be made before merging.

Comment (elisachisari) [reply src/ccl_massfunc.c]: Agree! I'll change it

File: src/ccl_massfunc.c [hunk 2]
```diff
@@ -574,7 +574,7 @@ double ccl_massfunc(ccl_cosmology *cosmo, double halomass, double a, double odel
 	  return NAN;
   }
 
-  if (halomass > 1e17 || halomass < 1e6){
+  if (log10(halomass) > cosmo->spline_params.LOGM_SPLINE_MAX || log10(halomass) < cosmo->spline_params.LOGM_SPLINE_MIN){
```
Comment (damonge) [inline src/ccl_massfunc.c]: Could we take log10(halomass) above and assign it to a variable to avoid having to calculate it twice?

Comment (elisachisari) [reply src/ccl_massfunc.c]: Done!

**Pull request merged**